### PR TITLE
 `azurerm_firewall_policy_rule_collection_group`: Adding `Mssql` for property `application_rule_collection`

### PR DESCRIPTION
**Pull request by xiaxyi**
fixing the issue mentioned: https://github.com/hashicorp/terraform-provider-azurerm/issues/17666

Mssql is supported, adding it to TF as well.

### CODE REVIEW

File: internal/services/firewall/firewall_policy_rule_collection_group_resource.go [hunk 1]
```diff
@@ -105,12 +105,15 @@ func resourceFirewallPolicyRuleCollectionGroup() *pluginsdk.Resource {
 										Optional: true,
 										Elem: &pluginsdk.Resource{
 											Schema: map[string]*pluginsdk.Schema{
+												// Mssql is supported but service team didn't include it in the current swagger, they are working on the fix in the next swagger release.
+												// todo: to use the sdk enum for this Mssql const
```
Comment (katbyte) [inline internal/services/firewall/firewall_policy_rule_collection_group_resource.go:109]: can you open a bug on rest specs for them to include it?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/firewall/firewall_policy_rule_collection_group_resource.go (hunk 1 above)
Comment (xiaxyi) [reply internal/services/firewall/firewall_policy_rule_collection_group_resource.go:109]: Sure, will do. I confirmed with them via email and they said they are working on the fix now.

**Comment by xiaxyi** [comment]
swagger issue:https://github.com/Azure/azure-rest-api-specs/issues/20215

**Comment by katbyte** [comment]
@xiaxyi - can you follow up with the swagger issue/service team to get the value added to swagger?

**Comment by birdnathan** [comment]
Any update on this? Hit by this issue this week.

**Comment by mbfrahry** [comment]
Ack! I missed this when I merged #18746. As they're the same, I'll go ahead and close this. Apologies for the confusion

**Pull request closed without merging**